src/shogun/optimization/NesterovMomentumCorrection.h

```diff
@@ -92,15 +92,18 @@ class NesterovMomentumCorrection: public MomentumCorrection
 	 *
 	 * @param gradient gradient
 	 * @param idx the index of the direction
+	 * @param delta update the change to correct descend direction
 	 * 
 	 * @return corrected descend direction
 	 */
-	virtual float64_t get_corrected_descend_direction(float64_t gradient, index_t idx)
+	virtual float64_t get_corrected_descend_direction(float64_t gradient, index_t idx,
+		float64_t& delta)
 	{
 		REQUIRE(idx>=0 && idx<m_previous_descend_direction.vlen,"The index (%d) is invalid\n", idx);
 		float64_t tmp=m_weight*m_previous_descend_direction[idx];
 		m_previous_descend_direction[idx]=tmp-gradient;
-		return (1.0+m_weight)*m_previous_descend_direction[idx]-tmp;
+		delta=m_previous_descend_direction[idx];
```

Comment:
@lisitsyn 
Yes.
For `NesterovMomentumCorrection`, the `return` is different from `delta`.